server/src/main/java/io/crate/execution/engine/JobLauncher.java

```diff
@@ -246,33 +246,43 @@ private void setupTasks(TransactionContext txnCtx,
             CompletableFutures.allAsList(directResponseFutures)
                 .whenComplete(BucketForwarder.asConsumer(pageBucketReceivers, bucketIdx, initializationTracker));
             bucketIdx++;
-            try {
-                // initializationTracker for localNodeOperations is triggered via SetBucketCallback
 
-                localTask.start();
-            } catch (Throwable t) {
-                accountFailureForRemoteOperations(operationByServer, initializationTracker, handlerPhaseAndReceiver, t);
-                return;
-            }
+            final int currentBucket = bucketIdx;
+            // initializationTracker for localNodeOperations is triggered via SetBucketCallback
+            localTask.start().whenComplete((result, err) -> {
+                if (err == null) {
```

Comment:
Can't we extract this code (with a currentBucket param)?

Reply:
The code paths in the callbacks are different. I don't think it's warranted to try to abstract this further.